configure.ac

```diff
@@ -1655,6 +1661,60 @@ if test "x$HAVE_ZLIB" = "xyes" -a "x$HAVE_ZLIB_H" = "xyes" -a "x$HAVE_COMPRESS2"
     EXTERNAL_FILTERS="${EXTERNAL_FILTERS}deflate(zlib)"
 fi
 
+## TODO - Probably this should be reworked or moved somewhere else
+
+# If the provided name corresponds to a non-empty variable,
+# create a macro with the given name and value. 
+# If the provided name does not exist, create a macro with the 
+# default value if one was provided, otherwise undefine the macro.
+m4_define([EXPORT_VARIABLE],
+  [if test -n "$$1"; then
+    AC_SUBST([$1], ["#define $1 $$1"])
+   else
+     if test -n "$2"; then
+       AC_SUBST([$1], ["#define $1 $2"])
+     else
+       AC_SUBST([$1], ["/*undef $1 */"])
+     fi
+   fi]
+)
+
+EXPORT_VARIABLE([H5_API_TEST_CONFIG_H], ["1"])
+
+# TODO: These flags are copied from a CMake equivalent H5_api_test_config.h.in 
+# and I haven't verified that the autotools define them
+EXPORT_VARIABLE([H5_API_TEST_HAVE_ASYNC], [])
```

Comment:
Realistically, this is probably the only #define out of the above and below that is actually needed, since I don't expect the Autotools to ever handle setting up of a server process like CMake used to. I think everything else can be deleted, at which point the m4 macros above also probably aren't needed.